src/dlr.cc

```diff
@@ -381,3 +365,94 @@ extern "C" int SetDLRCustomAllocatorMemalign(DLRMemalignFunctionPtr custom_memal
   DLRAllocatorFunctions::SetMemalignFunction(custom_memalign_fn);
   API_END();
 }
+
+extern "C" int CreateDLRModelFromGraphRuntime(DLRModelHandle* handle, const char* lib_path,
+                                              const char* graph, const char* params,
+                                              size_t params_len, const char* metadata, int dev_type,
+                                              int dev_id) {
+  API_BEGIN();
+  DLContext ctx;
+  ctx.device_type = static_cast<DLDeviceType>(dev_type);
+  ctx.device_id = dev_id;
+  std::string model_lib = FixWindowsDriveLetter(lib_path);
+  std::string param_str(params, params_len);
+  *handle = new TVMModel(model_lib, graph, &param_str, metadata, ctx);
+  API_END();
+}
+
+extern "C" int CreateDLRModelFromVMRuntime(DLRModelHandle* handle, const char* lib_path,
+                                           const char* relay_exec, size_t relay_len,
+                                           const char* metadata, int dev_type, int dev_id) {
+  API_BEGIN();
+  DLContext ctx;
+  ctx.device_type = static_cast<DLDeviceType>(dev_type);
+  ctx.device_id = dev_id;
+  std::string model_lib = FixWindowsDriveLetter(lib_path);
+  std::string relay_str(relay_exec, relay_len);
+  *handle = new RelayVMModel(model_lib, relay_str, metadata, ctx);
+  API_END();
+}
+
+extern "C" int SetDLRInputTensor(DLRModelHandle* handle, const char* name, void* dltensor) {
+  API_BEGIN();
+  DLRModel* dlr_model = static_cast<DLRModel*>(*handle);
+  CHECK(strcmp(dlr_model->GetBackend(), "tvm") == 0 ||
+        strcmp(dlr_model->GetBackend(), "relayvm") == 0)
+      << "model is not a TVMModel or RelayVMModel. Found '" << dlr_model->GetBackend()
+      << "' but expected 'tvm' or 'relayvm'";
+
+  DLTensor* tensor = static_cast<DLTensor*>(dltensor);
+  if (strcmp(dlr_model->GetBackend(), "tvm") == 0) {
```

Comment:
nit:  use strncmp to handle non-null terminated strings safely

Reply:
they are null terminated: 
https://github.com/neo-ai/neo-ai-dlr/blob/master/src/dlr_tvm.cc#L237
And there are lots of other strcmp uses elsewhere. Do we need to change this now or can we do it globally for DLR in another PR?

Reply:
Follow up is fine

Reply:
@kkoppolu1 i just pushed a change that uses enum instead of strings for this logic. so we're addressing your comment in an other way.

Reply:
nice!